FirebaseAppCheck/Interop/Public/FirebaseAppCheckInterop/FIRAppCheckInterop.h

```diff
@@ -23,7 +23,7 @@ NS_ASSUME_NONNULL_BEGIN
 NS_SWIFT_NAME(AppCheckTokenHandlerInterop)
 typedef void (^FIRAppCheckTokenHandlerInterop)(id<FIRAppCheckTokenResultInterop> tokenResult);
 
-NS_SWIFT_NAME(AppCheckInterop) @protocol FIRAppCheckInterop
+NS_SWIFT_NAME(AppCheckInterop) @protocol FIRAppCheckInterop<NSObject>
```

Comment:
what does this diff do?

Reply:
Good question. I found that when writing `FIRAppCheckInteropAPITests.m` I couldn't call [`respondsToSelector:`](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418583-respondstoselector?language=objc) on the `id<FIRAppCheckInterop>` because the protocol didn't conform to [`NSObject`](https://developer.apple.com/documentation/objectivec/1418956-nsobject?language=objc). I was doing this to show that `getLimitedUseTokenWithCompletion:` was an optional method in the `FIRAppCheckInterop` protocol.

Note: It looks like none of our SDKs call `getLimitedUseTokenWithCompletion:` yet so I could revert the change in this PR to keep it more minimalistic.

Reply:
Interesting find. I'm ok with whatever you decide.

One alt. way to show that getLimitedUseTokenWithCompletion: is an optional method in the FIRAppCheckInterop protocol may be to make a new type that doesn't implement the API. If the API was optional, the type wouldn't built because it doesn't implement it.

Reply:
That would indeed show that it's optional but my intent in `FIRAppCheckInteropAPITests.m` was to test the usage of `FIRAppCheckInterop` as a consumer of the API (and other SDKs wouldn't be making a type that conforms to `FIRAppCheckInterop`, other than for making test fakes, since they'd get an instance from the component system).